bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,424 @@
+    cat <<'EOF'
+{ "application": {
+    "extensions": {
+        "allowed": {
+          "enabled": false,
+          "list": ["less-than:<", "greater:>", "and:&", "d-quote:\"", "emoji:😀"]
+        },
+        "installed": {
+            "key-with-less-than: <": true,
+            "key-with-ampersand: &": true,
+            "key-with-greater-than: >": true,
+            "key-with-emoji: 🐤": false
+        }
+    }
+  },
+  "containerEngine": {
+    "name": "small-less-<-than"
+  }
+}
+EOF
+}
+export -f json_with_special_chars
+
+assert_full_setting_registry_output() {
+    local hive=$1
+    local type=$2
+    assert_success
+    assert_output --partial "Windows Registry Editor Version 5.00"
+    assert_output --partial "[$hive\\SOFTWARE\\Policies\\Rancher Desktop\\$type\\application]"
+    assert_output --partial '"debug"=dword:1'
+    assert_output --partial "[$hive\\SOFTWARE\\Policies\\Rancher Desktop\\$type\\images]"
+    assert_output --partial '"namespace"="change-to-k8s.io"'
+}
+
+assert_full_setting_plist_output() {
+    assert_success
+    assert_output --partial '<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">'
+    assert_output --partial '<plist version="1.0">'
+    # this next line makes sense only after <key>namespace</key>
+    assert_output --partial '<string>change-to-k8s.io</string>'
+}
+
+@test 'generates registry output for hklm/defaults' {
+    run rdctl create-profile --output reg --from-settings
+    assert_full_setting_registry_output HKEY_LOCAL_MACHINE defaults
+
+    run rdctl create-profile --output reg --hive=hklm --from-settings
+    assert_full_setting_registry_output HKEY_LOCAL_MACHINE defaults
+
+    run rdctl create-profile --output reg --hive=HKLM --type=Defaults --from-settings
+    assert_full_setting_registry_output HKEY_LOCAL_MACHINE defaults
+
+    run rdctl create-profile --output reg --type=DEFAULTS --from-settings
+    assert_full_setting_registry_output HKEY_LOCAL_MACHINE defaults
+}
+
+@test 'generates registry output for hklm/locked' {
+    run rdctl create-profile --output reg --hive=Hklm --type=Locked --from-settings
+    assert_full_setting_registry_output HKEY_LOCAL_MACHINE locked
+
+    run rdctl create-profile --output reg --type=LOCKED --from-settings
+    assert_full_setting_registry_output HKEY_LOCAL_MACHINE locked
+}
+
+@test 'generates registry output for hkcu/defaults' {
+    run rdctl create-profile --output reg --hive=Hkcu --from-settings
+    assert_full_setting_registry_output HKEY_CURRENT_USER defaults
+
+    run rdctl create-profile --output reg --hive=hkcu --type=Defaults --from-settings
+    assert_full_setting_registry_output HKEY_CURRENT_USER defaults
+}
+
+@test 'generates registry output for hkcu/locked' {
+    run rdctl create-profile --output reg --hive=HKCU --type=locked --from-settings
+    assert_full_setting_registry_output HKEY_CURRENT_USER locked
+}
+
+assert_check_registry_output() {
+    local mainOutput=$1
```

Comment:
I'm not sure if we need to pass `"$output"` as a parameter. We run `assert_success`, which assumes `$status` is still set, so why no assume `$output` as well? It is even part of the function name `assert_check_registry_output`. And it is what we are doing in `assert_full_setting_plist_output`.

Reply:
I did that because I was gettings shellcheck warnings:
```
SC2030 (info): Modification of output is local (to subshell caused by @bats test).
SC2031 (info): output was modified in a subshell. That change might be lost.
```

Passing `$output` copied it into a variable and removed the warnings. Now I need shellcheck directives.